Carson inventory map

xsedim opened the pull request:
Inventory map is now fully functional:
- Player map marker moves to be in the room Link is currently in
- Rooms show up on the map as they're discovered by Link

<br>

Also fixed the bug where Link could move around while paused which broke a lot of things if you abused it

RobertEGreenslade12 commented on MonoZelda/Dungeons/Parser/Data/DungeonConstants.cs:
```diff
@@ -82,4 +82,50 @@ public static class DungeonConstants
+            new Point(1, 0),
+        };
+    }
+    public static string GetRoomMapSprite(Point coord) {
+        return coord switch {
+            Point(3, 5) => "map_room_NEW",
+            Point(2, 5) => "map_room_E",
+            Point(3, 4) => "map_room_NS",
+            Point(4, 5) => "map_room_W",
+            Point(3, 3) => "map_room_NESW",
+            Point(2, 3) => "map_room_NE",
+            Point(3, 2) => "map_room_NESW",
+            Point(4, 3) => "map_room_NW",
+            Point(2, 2) => "map_room_ESW",
+            Point(1, 2) => "map_room_E",
+            Point(3, 1) => "map_room_NS",
+            Point(4, 2) => "map_room_ESW",
+            Point(5, 2) => "map_room_NW",
+            Point(5, 1) => "map_room_ES",
+            Point(6, 1) => "map_room_W",
+            Point(3, 0) => "map_room_SW",
+            Point(2, 0) => "map_room_EW",
+            Point(1, 0) => "map_room_E",
+            _ => null,
```
Should these instead be the room number?

xsedim replied:
It's a utility method, so I just wrote for the data conversion I needed. If a room name to room sprite name utility method is needed in the future then it could be added, but it would have just been more work then necessary for my purposes in this branch.

RobertEGreenslade12 commented on MonoZelda/Link/PlayerSpriteManager.cs:
```diff
@@ -3,6 +3,7 @@
 using MonoZelda.Commands.GameCommands;
 using System.Collections.Generic;
 using MonoZelda.Dungeons;
+using System.Diagnostics;
```
Probably don't need this

RobertEGreenslade12 reviewed (changes requested):
I like the changes, but I think we could make the canMove flag more general to PlayerEnabled. We could then block all player interaction. Currently in this branch, I can still attack and use projectiles while in inventory. I believe all player commands should be blocked while in inventory. Otherwise, we allow the user to pause when an enemy is in front of them, and then proceed to kill it in pause state. Let me know if this makes sense. It should just be renaming canMove to playerEnabled and using that to block all player commands instead of just move.

RobertEGreenslade12 commented:
Make sure to re-request the review to unblock merging this branch when you have made the fixes.

xsedim commented:
@RobertEGreenslade12 I implemented the functionality you described but I did it in a different way. Rather than being on the player end, now when the game is paused the current `RoomScene` replaces `CommandManager's` player action commands with empty commands so that nothing happens when that command is invoked. In the future, these can be swapped out with different commands so that you can select items in the inventory.

Also fixed the bug where an error is thrown if you try to use a projectile when no projectile is selected.
